pkgs/shells/fish/default.nix

```diff
@@ -152,11 +159,20 @@ let
       # the shell's actual version (and what it displays when running `fish
       # --version`), as well as the local documentation for all builtins (and
       # maybe other things).
-      url = "https://github.com/fish-shell/fish-shell/releases/download/${version}/${pname}-${version}.tar.xz";
-      hash = "sha256-YUyfVkPNB5nfOROV+mu8NklCe7g5cizjsRTTu8GjslA=";
+      url = "https://github.com/fish-shell/fish-shell/releases/download/${finalAttrs.version}/${finalAttrs.pname}-${finalAttrs.version}.tar.xz";
+      hash = "sha256-U0M04Q+FciIU6dr/gqV8w1ASNVI/FvjxMcI0Tk7Jjac=";
+    };
+
+    cargoDeps = rustPlatform.fetchCargoVendor {
+      inherit (finalAttrs) src;
```

Comment:
Isn't this [IFD which is against nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#import-from-derivation).

Reply:
Which you could use the cargoLock method:
```
  cargoDeps = rustPlatform.importCargoLock {
    lockFile = ./Cargo.lock;
    outputHashes = {
      # Git Hashes here
    };
  };
  ```

Reply:
> Isn't this [IFD which is against nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#import-from-derivation)?

In case this comment was genuine, no this isn't IFD. We are simply using a value of a nix variable known at eval time. IFD is when you import things at eval time from the resulting build output of a derivation, in which case nix would actually need to build that derivation during eval. But that's not what's happening here.